clcache.py

```diff
@@ -164,33 +164,43 @@ def clean(self, stats, maximumSize):
         except ImportError:
             walker = os.walk
 
-        objects = [os.path.join(root, "object")
-                   for root, folder, files in walker(self.objectsDir)
-                   if "object" in files]
+        objectInfos = []
+        for x in xrange(16):
+            try:
+                objects = [os.path.join(root, "object")
+                       for root, folder, files in walker(self.objectsDir)
+                       if "object" in files]
+
+                objectInfos = [(os.stat(fn), fn) for fn in objects]
```

Comment:
I am not exactly sure how hard the mess can be, but

 1. Find that cache entry exists
 2. Restore from cache
 3. Update stats

are not atomic and the deletion might happen anywhere in between.

> people who are not aware of clcache (like sys admins) will just log onto build servers, see the huge folder and delete it.

Isn't that an issue of company structures, responsibilities and competences? Sounds like you have much bigger issues then build fails when you run in this kind of situations. I mean either this guy knows the software that is running or he/she should not be allowed to delete stuff.